### PR TITLE
Fixed sample duration strange cases

### PR DESCRIPTION
**Pull request by brabebhin**
In some cases, ffmpeg report wrong sample durations for packages. This helps deal with that situation

### CODE REVIEW

**Comment by lukasf** [comment]
Looks fine.

**Comment by mcosmin222** [comment]
@khouzam , to which case are you referring exactly? The calculation of block_align?

File: FFmpegInterop/Source/UncompressedAudioSampleProvider.cpp [hunk 1]
```diff
@@ -134,7 +138,14 @@ MediaStreamSample^ UncompressedAudioSampleProvider::GetNextSample()
 			{
 				finalPts = pts;
 			}
-			finalDur += dur;
+			if (dur == 0 && dataWriter->UnstoredBufferLength != 0)
+			{
+				finalDur = (dataWriter->UnstoredBufferLength / avgBytesPerSec) * 10000000;
```
Comment (lukasf) [inline FFmpegInterop/Source/UncompressedAudioSampleProvider.cpp]: I think @khouzam means this line. Maybe better do it like this?

`finalDur = ((long long)dataWriter->UnstoredBufferLength * 10000000) / avgBytesPerSec`

I am not sure if the long long cast is neccessary, but we need to be sure that we don't get overflow.

Comment (mcosmin222) [reply FFmpegInterop/Source/UncompressedAudioSampleProvider.cpp]: I will change all of them this weekend. I haven't had the time to do much testing and development on this lately :(

**Comment by lukasf** [comment]
Somehow, your last changeset changed the whole file. Did you perhaps accidentially change line endings? Not such a big deal for me, but it seems to break GitHub diff tool.

Anyways, changes look fine to me.

**Comment by mcosmin222** [comment]
Strange. I did edit it in the online editor, maybe that is what caused the issue.

**Pull request closed without merging**